connectivity/cellular/source/framework/AT/AT_CellularContext.cpp

```diff
@@ -50,15 +50,15 @@ using namespace std::chrono_literals;
 AT_CellularContext::AT_CellularContext(ATHandler &at, CellularDevice *device, const char *apn, bool cp_req, bool nonip_req) :
     _current_op(OP_INVALID), _dcd_pin(NC), _active_high(false), _cp_req(cp_req), _is_connected(false), _at(at)
 {
-    tr_info("New CellularContext %s (%p)", apn ? apn : "", this);
+    tr_info("New CellularContext %s (%p)", apn ? apn : "", (void *)this);
```

Comment:
i agree with you but my intetion is to reduce the warning spam when turning on pedandic and wall
this is a minor change and i assumed to be not as invasive as to push/pop the compiler warning
but i can change it

Reply:
`-pedantic` is what it says it is. It warns about every single thing that is not standard C++, unless it's something clearly delineated as implementation-defined (eg by having lots of underscores).

I don't believe it's always worth modifying code purely for the sake of satisfying a totally pedantic compiler. There is sometimes a trade-off.

And you do have finer-grained control - you should be able to request `-pedantic -Wall` and then tune it by adding more options to silence a few specific warnings that those enable

Do you want the Mbed OS code to be clean because you want your own Mbed OS app code to be fully pedantic checked, so are setting that in the compiler profile? If so, then this falls into the area of having separate build options for different parts of the tree. I guess that might be possible soon as part of the CMake work?